rinstall.sh

```diff
@@ -36,50 +36,73 @@ case $(dirname "$target") in
 		;;
 esac
 
+# If a source file doesn't exist, try to download it
 if [ ! -f "$1" ]; then
-	umask 044
-	source="$(basename "$1" | tr -C '0-9a-zA-Z_' '_')$(mktemp XXXXXX)"
-        case $(uname) in
+	# The source file doesn't exist at this point and it shouldn't be
+	# an absolute path in this case
+	spath=$(dirname $source)
+	case $spath in
+		/*)
+			>&2 echo "rinstall: $source should not be an absolute path" \
+				     "when resides on a remote HTTP server"
```

Comment:
Let's try this without the line continuation. I do think it's a good habit to avoid long lines, but sometimes a long line reads better.

Suggested edit to match the pattern *utility: error message: [details]*

> rinstall: absolute path not permitted when fetching over HTTP: $source

Reply:
I also didn't like this long message, but couldn't come up with a good error message.